File: src/main/java/com/itau/elegibilidade/inss/entrypoint/RegraRequest.java
package com.itau.elegibilidade.inss.entrypoint;

public class RegraRequest {

    Integer codigoConvenio;
    Integer codigoDominio;
    Integer codigoModalidade;

    public Integer getCodigoConvenio() {
        return codigoConvenio;
    }

    public void setCodigoConvenio(Integer codigoConvenio) {
        this.codigoConvenio = codigoConvenio;
    }

    public Integer getCodigoDominio() {
        return codigoDominio;
    }

    public void setCodigoDominio(Integer codigoDominio) {
        this.codigoDominio = codigoDominio;
    }

    public Integer getCodigoModalidade() {
        return codigoModalidade;
    }

    public void setCodigoModalidade(Integer codigoModalidade) {
        this.codigoModalidade = codigoModalidade;
    }
}

File: src/main/kotlin/com/itau/elegibilidade/inss/entrypoint/RegraRequest.kt
package com.itau.elegibilidade.inss.entrypoint;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.math.BigDecimal

@JsonIgnoreProperties(ignoreUnknown = true)
data class RegraRequest(
        val valorParcela: BigDecimal? = null,
        val valorSolicitado: BigDecimal? = null,
        val codigoBanco: Int? = null,
        val numeroAgencia: Int? = null,
        val numeroConta: Int? = null,
        val numeroDigitoConta: Int? = null,
        val codigoMeioPagamento: String? = null,
        val indicadorContaPrincipal: Boolean? = null,
        val siglaUfBeneficio: String? = null,
        val codigoConvenio: Int? = null,
        val codigoDominio: Int? = null,
        val codigoModalidade: Int? = null
)
